src/extension/wps/wps-core/src/main/java/org/geoserver/wps/ppio/EWKTPPIO.java

```diff
@@ -0,0 +1,75 @@
+/* (c) 2023 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wps.ppio;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Pattern;
+import org.apache.commons.io.IOUtils;
+import org.geotools.geometry.jts.WKTReader2;
+import org.geotools.referencing.CRS;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.io.WKTWriter;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+/**
+ * Used to represent geometries in EWKT format Reads geometry and crs from string and sets this crs
+ * to created geometry
+ */
+public class EWKTPPIO extends WKTPPIO {
```

Comment:
Looks like you have overridden every single method from WKTPPIO, is there any reason left to use inheritance?

Reply:
Changed to CDataPPIO and reverted changes to WKTPPIO